duckduckgo/duckduckgo-privacy-extension
# Youtube Click To Load - UI Implementation

**Reviewer:**
**CC**: @kzar 

## Description:

1. This PR adds the UI implementation for YouTube Click to Load.
2. This also adds "Enable YouTube Previews" to the Settings page.
3. It also syncs the UI between Facebook CTL and YouTube CTL (adds "Share Feedback" link and updates the modal UI for both CTL)
4. This does not fully implements the working feature of blocking and displaying the YT previews. This will be done in a following PR.

Test
Build steps: https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/develop/CONTRIBUTING.md#building-the-extension

## Steps to test this PR:

1. Access any website with embedded YouTube content on Chrome or Firefox. You can also use our test page for this https://privacy-test-pages.glitch.me/privacy-protections/youtube-click-to-load/
2. All YouTube videos show be available
3. Follow [build steps](https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/develop/CONTRIBUTING.md#building-the-extension) and load the extension into Chrome or Firefox and refresh your test page.
4. All YouTube videos should be blocked and display the blocked content dialog with a "Share Feedback" link bellow it.
    1. Press "Play" should play the video
    2. Press "Previews disabled for additional privacy" should open a modal
    3. Press the "X" or "No thanks" should close the modal with no changes
    4. Press the "Enable Previews" button should close the modal. (Replacing blocked content with preview not implemented in this PR - please refresh the page)
    5. Press "Learn More" should open Help Pages for embedded content.
5. Enable YT Previews and refresh the page. With YT Previews enabled, all embedded videos should display the video preview with "Share Feedback" link bellow it.
    1. Press "Play" should play the video.
    2. Press "Learn More" should open Help Pages for embedded content.
    3. Press the "Previews enabled" and refresh the page should show the blocked content ui again on all videos. (Replacing the preview with the blocked content ui not implemented in this PR - please refresh the page)
6. ...


## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
